frontend/src/store/objectStore.ts

```diff
@@ -123,15 +129,24 @@ export const useObjectStore = defineStore('object', () => {
 
         let response = Array<COMSObject>();
         if (uniqueIds.length) {
+          // If metadata specified, search for objects with matching metadata
+          const headers: Record<string, string> = {};
+          if (metadata?.length) {
+            for (const meta of metadata) {
+              headers[`x-amz-meta-${meta.key}`] = meta.value;
+            }
+          }
+
           response = (await objectService.searchObjects({
             bucketId: params.bucketId ? [params.bucketId] : undefined,
             objectId: uniqueIds,
```

Comment:
not from this PR but i'm not sure this fetahObjects() process is feasible.
the list of objectId's could be way too long to fit in a url, no?

Reply:
Yeah it's always put every object ID in there to get the table from the permissions from what I've ever seen. Not sure if that's a sustainable solution yeah.
COMS searching on permissions and doing that in SQL could be a better option 🤷